clients/src/main/java/io/strimzi/kafka/KafkaConsumerClient.java

```diff
@@ -103,15 +103,20 @@ private void checkAndReceiveMessages() {
     public void consumeMessages() {
         ConsumerRecords<String, String> records = consumer.poll(Duration.ofMillis(Long.MAX_VALUE));
 
-        for (ConsumerRecord<String, String> record : records) {
-            LOGGER.info("Received message:");
-            LOGGER.info("\tpartition: {}", record.partition());
-            LOGGER.info("\toffset: {}", record.offset());
-            LOGGER.info("\tvalue: {}", record.value());
-            if (record.headers() != null) {
-                LOGGER.info("\theaders: ");
-                for (Header header : record.headers()) {
-                    LOGGER.info("\t\tkey: {}, value: {}", header.key(), new String(header.value()));
+        for (ConsumerRecord<String, String> consumerRecord : records) {
+            if (configuration.getOutputFormat().equalsIgnoreCase("json")) {
+                String jsonLog = KafkaConsumerRecord.parseKafkaConsumerRecord(consumerRecord).toJsonString();
+                LOGGER.info("Received message: {}", jsonLog);
+            } else {
+                LOGGER.info("Received message:");
+                LOGGER.info("\tpartition: {}", consumerRecord.partition());
+                LOGGER.info("\toffset: {}", consumerRecord.offset());
+                LOGGER.info("\tvalue: {}", consumerRecord.value());
+                if (consumerRecord.headers() != null) {
+                    LOGGER.info("\theaders: ");
+                    for (Header header : consumerRecord.headers()) {
+                        LOGGER.info("\t\tkey: {}, value: {}", header.key(), new String(header.value()));
+                    }
```

Comment:
For the `plain` type of output message, I would add a method into the `KafkaConsumerRecord` as well and maybe add one more method for logging into the object, to keep it on one place and based on the settings just output the result.

So the code can then look like this:

```
for (ConsumerRecord<String, String> consumerRecord : records) {
    KafkaConsumerRecord record = KafkaConsumerRecord.parseKafkaConsumerRecord(consumerRecord);
    record.logMessage();
}
```

Reply:
Sounds pretty reasonable. Thanks!